### PR TITLE
feat: OPA executor

### PR DESCRIPTION
**Pull request by suchapalaver**
#### Interfaces for OPA policy execution and loading     

`OpaExecutor` takes a Chronicle `AuthId` identity and an arbitrary JSON context to evaluate against an OPA policy, which, in the `WasmtimeOpaExecutor` implementation, is compiled to `.wasm`.

`PolicyLoader`'s `SawtoothPolicyLoader` implementation loads an OPA policy rule compiled to `.wasm` from a Sawtooth address, while its `CliPolicyLoader` implementation provides support to load an OPA policy `.wasm` file via CLI arguments for a file path and policy entrypoint.

---
[CHRON-155](https://blockchaintp.atlassian.net/browse/CHRON-155?atlOrigin=eyJpIjoiNGJhZDA4NDM3NmQ2NDdhYTk1M2U5MDgyYzQ0NmFiMzciLCJwIjoiaiJ9)

Signed-off-by: Joseph Livesey <joseph.livesey@btp.works>

[CHRON-155]: https://blockchaintp.atlassian.net/browse/CHRON-155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: crates/common/src/opa_executor.rs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+impl SawtoothPolicyLoader {
+    pub fn new(address: &url::Url) -> Self {
+        let (tx, _rx) = ZmqMessageConnection::new(address.as_str()).create();
+        Self {
+            tx, _rx
+        }
+    }
+
+    #[instrument(level = "info", skip(self), ret)]
+    async fn get_policy(&mut self, address: &str) -> Result<Vec<u8>, SawtoothPolicyLoaderError> {
```
[the remark this answers is not in the document]
Comment (ryanbtp) [reply crates/common/src/opa_executor.rs:58]: But only use the address bit for now

File: crates/common/src/identity.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use k256::ecdsa::{signature::Signer, Signature, SigningKey, VerifyingKey};
+
+custom_error! {pub IdentityError
+    KeyStore{source: SignerError} = "Invalid key store directory {source}",
+    SerdeJson{source: serde_json::Error } = "Malformed JSON {source}",
+    Signing{source: k256::ecdsa::Error} = "Signing error {source}",
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
+pub enum AuthId {
```
[the remark this answers is not in the document]
Comment (suchapalaver) [reply crates/common/src/identity.rs:16]: Will address this in a separate PR

**Comment by mtbc** [comment]
I am curious about the source of the wasm files.

File: crates/common/src/opa_executor.rs [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+
+        let result = executor.evaluate(&id, context).await?;
+
+        assert!(result);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn opa_executor_unauthorized_id() -> Result<(), OpaExecutorError> {
```
Comment (mtbc) [inline crates/common/src/opa_executor.rs:334]: :+1: Great to test these failure cases.

**Review by ryanbtp — DISMISSED** [review]

**Review by ryanbtp — APPROVED** [review]

**Pull request merged**